src/aead/chacha20_poly1305.rs

```diff
@@ -51,12 +55,20 @@ fn chacha20_poly1305_seal(
     aad: Aad<&[u8]>,
     in_out: &mut [u8],
     cpu_features: cpu::Features,
-) -> Tag {
+) -> Result<Tag, error::Unspecified> {
     let chacha20_key = match key {
         aead::KeyInner::ChaCha20Poly1305(key) => key,
         _ => unreachable!(),
     };
 
+    if in_out.len() > MAX_IN_OUT_LEN {
+        return Err(error::Unspecified);
+    }
```

Comment:
This code path only occurs with a (practically) impossibly-long input, so we can't test it.